btfgen.c

```diff
@@ -193,6 +219,49 @@ static int btf_reloc_id_add(struct btf_reloc_info *info, unsigned int old, unsig
 	return hashmap__add(info->ids_map, uint_as_hash_key(old), uint_as_hash_key(new));
 }
 
+/* Return pointer to btf_reloc_type by id */
+static struct btf_reloc_type *btf_reloc_get_type(struct btf_reloc_info *info, int id) {
+	struct btf_reloc_type *type = NULL;
+
+	if (!hashmap__find(info->types, uint_as_hash_key(id), (void **)&type))
+		return ERR_PTR(-ENOENT);
+
+	return type;
+}
+
+static int bpf_reloc_type_add_member(struct btf_reloc_info *info,
+				     struct btf_reloc_type *reloc_type,
+				     struct btf_member *btf_member, int idx) {
+	int err;
+	struct btf_reloc_member *reloc_member;
+
```

Comment:
I think you could also check `added_by_all` here. If it's set then nothing else we should do.

Reply:
Not sure I follow this one. Checking for the reloc_type->added_by_all here won't allow a second, third and forth member to be added, for example, as we're adding them all. 

Or is this something else ?

Reply:
Exactly. If this type has `added_by_all` it's completed (all members are already added).